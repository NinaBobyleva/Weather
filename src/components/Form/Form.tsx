import React, { useState } from "react";

export function Form({
  setNewCityName: setNewCityName,
}: {
  setNewCityName: React.Dispatch<React.SetStateAction<string>>;
}) {
  const [cityNameInput, setCityNameInput] = useState("");

  const handleNewCityForm = (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault();
    setNewCityName(cityNameInput);
    setCityNameInput("");
  };

  return (
    <form
      onSubmit={handleNewCityForm}
      className="flex flex-wrap xl:flex-nowrap justify-center gap-4 py-4 sm:py-8 md:py-12"
    >
      <input
        onChange={(e) => setCityNameInput(e.target.value)}
        type="search"
        className="border-2 rounded-2xl px-5 py-2 outline-none w-[320px] sm:w-[500px] md:w-[700px]"
        value={cityNameInput}
        placeholder="Введите город"
      />

      <button
        type="submit"
        className="bg-slate-400 px-7 py-2 rounded-2xl"
      >
        Найти
      </button>
    </form>
  );
}
