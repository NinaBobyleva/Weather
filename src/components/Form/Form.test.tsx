import * as renderer from "react-test-renderer";
import { Form } from "./Form";

describe("Компонент Form", () => {
  it("Успешный рендер компонента Form", () => {
    const setNewCityName = jest.fn();
    const component = renderer
      .create(<Form setNewCityName={setNewCityName} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});