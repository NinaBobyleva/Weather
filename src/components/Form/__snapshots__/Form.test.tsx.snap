// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент Form Успешный рендер компонента Form 1`] = `
<form
  className="flex flex-wrap xl:flex-nowrap justify-center gap-4 py-8 sm:py-8 md:py-12"
  onSubmit={[Function]}
>
  <input
    className="border-2 rounded-2xl px-5 py-2 outline-none w-[320px] sm:w-[500px] md:w-[700px]"
    onChange={[Function]}
    placeholder="Введите город"
    type="search"
    value=""
  />
  <button
    className="bg-slate-400 px-7 py-2 rounded-2xl"
    type="submit"
  >
    Найти
  </button>
</form>
`;
